linguatec_lexicon/views.py

```diff
@@ -75,11 +79,182 @@ def validate(self, xlsx_file):
         return out
 
 
+class ImportDataView(TemplateView):
+    template_name = "linguatec_lexicon/importdata.html"
+
+    def create_imports_info(self, input_file, type):
+        ii = ImportsInfo.objects.create(status='created', input_file=input_file, type=type)
+        return ii.pk
+
+    def post(self, request, *args, **kwargs):
+        context = self.get_context_data(**kwargs)
+
+        type_import = str(request.POST.get('type_import'))
+
+        if type_import == 'data' or type_import == 'variation':
+            form = ValidatorForm(request.POST, request.FILES)
+        if type_import == 'gramcats':
+            form = CSVValidatorForm(request.POST, request.FILES)
+
+        if form.is_valid():
+            file = form.cleaned_data['input_file']
+            imports_info_id = self.create_imports_info(file, type_import)
+
+            # store uploaded file as a temporal file
+            if type_import == 'data' or type_import == 'variation':
+                tmp_fd, tmp_file = tempfile.mkstemp(suffix='.xlsx')
+            if type_import == 'gramcats':
+                tmp_fd, tmp_file = tempfile.mkstemp(suffix='.csv')
+            f = os.fdopen(tmp_fd, 'wb')  # open the tmp file for writing
+            f.write(file.read())  # write the tmp file
+            f.close()
+
+            if type_import == 'data':
+                lexicon_code = str(request.POST.get('lexicon_code'))
+                lexicon_id = Lexicon.objects.get(src_language=lexicon_code[:2], dst_language=lexicon_code[3:]).pk
```

Comment:
I suggest to create a custom method on LexiconManager to retrieve a Lexicon by its code. I mean something like that:
```python
# models.py
class LexiconManager(models.Manager):
    def get_by_code(self, code):
        src_lang = code[:2]
        dst_lang = code[3:]
        return self.get(src_language=src_lang, dst_language=dst_lang)

class Lexicon(models.Model):
    ....
    objects = LexiconManager()
```

You can get more information checking the docs of [custom model managers](https://docs.djangoproject.com/en/3.1/topics/db/managers/#custom-managers).

Reply:
Added and changed all the code to use .get_by_code instead of appliyng and rewritting this function in all the code, thank you for the info and aportation